code/_onclick/hud/hud.dm

```diff
@@ -113,179 +117,86 @@ var/datum/global_hud/global_hud = new()
 	var/obj/screen/r_hand_hud_object
 	var/obj/screen/l_hand_hud_object
 	var/obj/screen/action_intent
-	var/obj/screen/move_intent
+	var/obj/screen/zone_select
+	var/obj/screen/pull_icon
+	var/obj/screen/throw_icon
+	var/obj/screen/module_store_icon
 
-	var/list/adding
-	var/list/other
-	var/list/obj/screen/hotkeybuttons
+	var/list/static_inventory = list() //the screen objects which are static
+	var/list/toggleable_inventory = list() //the screen objects which can be hidden
+	var/list/obj/screen/hotkeybuttons = list() //the buttons that can be used via hotkeys
+	var/list/infodisplay = list() //the screen objects that display mob info (health, alien plasma, etc...)
+	var/list/screenoverlays = list() //the screen objects used as whole screen overlays (flash, damageoverlay, etc...)

```

Comment:
In order to make f12 toggle work for all mobs but not have to make a snowflake version of proc/show_hud() for each mob type, I added all obj/screen of all mob huds into these 5 lists (which is what list/adding and list/other were already doing, I just renamed them to static_inventory and toggleable_inventory to be less confusing, and added three more lists). This way all mob call the same version of show_hud() which uses these 5 lists directly instead of obj/screen's individually.